tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -761,24 +783,37 @@ void EnqueueWrapCommand::process() {
 EnqueueCommandType EnqueueWrapCommand::type() { return this->type_; }
 
 // CommandQueue section
-CommandQueue::CommandQueue(Device* device, uint32_t command_queue_channel): manager(*device->sysmem_manager) {
+CommandQueue::CommandQueue(Device* device, uint32_t command_queue_channel): manager(SystemMemoryManager::instance(device)) {
     this->device = device;
     this->command_queue_channel = command_queue_channel;
 
     chip_id_t mmio_device_id = tt::Cluster::instance().get_associated_mmio_device(device->id());
     this->command_queue_channel_size = tt::Cluster::instance().get_host_channel_size(mmio_device_id, tt::Cluster::instance().get_assigned_channel_for_device(mmio_device_id)) / device->producer_cores().size();
 
-    uint32_t channel_idx = 0;
-    const auto& dispatch_cores = device->consumer_cores();
-    if (auto it = std::find_if(dispatch_cores.begin(), dispatch_cores.end(),
-        [&channel_idx, &command_queue_channel](const CoreCoord& coord) {
-            return (channel_idx++ == command_queue_channel);
+    const auto get_used_core = [this, &command_queue_channel](const set<CoreCoord>& cores, CoreCoord& target) {
+        uint32_t channel_idx = 0;
+        if (auto it = std::find_if(cores.begin(), cores.end(),
+            [&channel_idx, &command_queue_channel](const CoreCoord& coord) {
+                return (channel_idx++ == command_queue_channel);
+            }
+        ); it != cores.end()) {
+            target = *it;
+        } else {
+            TT_THROW("Could not find a dispatch core for the provided channel");
         }
-    ); it != dispatch_cores.end()) {
-        this->dispatch_core = *it;
-    } else {
-        TT_THROW("Could not find a dispatch core for the provided channel");
-    }
+    };
+
+    get_used_core(device->producer_cores(), this->producer_core);
+    get_used_core(device->consumer_cores(), this->consumer_core);
+
+    // Start command queue
+    this->reset();
```

Comment:
GetCommandQueue is still creating a new CommandQueue object each time it is called so each time, we will launc h the dispatch kernels and write to dispatch cores L1?

Reply:
For now, yes. I'll need to update GetCommandQueue to not do this anymore, since it's not lightweight anymore.

Reply:
Updated.